antiplag/antiplag/views.py

```diff
@@ -28,22 +30,45 @@ class FileDetail(APIView):
     parser_classes = (MultiPartParser, FormParser)
 
     def post(self, request, *args, **kwargs):
-
-        submission_serializer = SubmissionSerializer(data=request.data)
+        submission_serializer = SubmissionSerializer(data={"status": Submission.SubmissionStatus.PENDING})
         if submission_serializer.is_valid():
             submission = submission_serializer.save()
-            for file in request.FILES.getlist("files"):
-                doc_serializer = DocumentSerializer(data={"file": file})
+
+            if request.META['HTTP_TYPE'] == Document.DocumentType.FILE:
+                for file in request.FILES.getlist("files"):
+                    text = Document.process_file(self, file)
+                    text_raw = Document.process_raw_text(self, text)
+                    language = Document.detect_language(self, text_raw)
+                    doc_serializer = DocumentSerializer(data={"file": file,
+                                                              "type": Document.DocumentType.FILE,
+                                                              "text": text,
+                                                              "text_raw": text_raw,
+                                                              "language": language
+                                                              })
+                    if doc_serializer.is_valid():
+                        document = doc_serializer.save()
+                        document.submission = submission
+                        document.save()
+                return Response(SubmissionSerializer(submission).data, status=status.HTTP_201_CREATED)
+            elif request.META['HTTP_TYPE'] == Document.DocumentType.TEXT:
+
+                text = request.POST.get('text', "no_text_attribute")
+                raw_text = request.POST.get('text', "no_text_attribute")
```

Comment:
Get only raw text and parse it using `Document. create_and_process_text()`. If no text is sent, use empty string instead. Or maybe it should just fail?